src/main/native/windows/util.cc

```diff
@@ -92,28 +92,27 @@ bool AutoAttributeList::Create(HANDLE stdin_h, HANDLE stdout_h, HANDLE stderr_h,
     return false;
   }
 
+  std::unique_ptr<AutoAttributeList> attr_list(
+      new AutoAttributeList(std::move(data), stdin_h, stdout_h, stderr_h));
   static constexpr size_t kHandleCount = 3;
-  std::unique_ptr<HANDLE[]> handles(new HANDLE[kHandleCount]);
-  handles[0] = stdin_h;
-  handles[1] = stdout_h;
-  handles[2] = stderr_h;
   if (!UpdateProcThreadAttribute(attrs, 0, PROC_THREAD_ATTRIBUTE_HANDLE_LIST,
-                                 handles.get(), kHandleCount * sizeof(HANDLE),
-                                 NULL, NULL)) {
+                                 attr_list->handles_.handle_array,
+                                 kHandleCount * sizeof(HANDLE), NULL, NULL)) {
     if (error_msg) {
       DWORD err = GetLastError();
       *error_msg = MakeErrorMessage(WSTR(__FILE__), __LINE__,
                                     L"UpdateProcThreadAttribute", L"", err);
     }
     return false;
   }
-  result->reset(new AutoAttributeList(&data, &handles));
+  *result = std::move(attr_list);
   return true;
 }
 
-AutoAttributeList::AutoAttributeList(std::unique_ptr<uint8_t[]>* data,
-                                     std::unique_ptr<HANDLE[]>* handles)
-    : data_(data->release()), handles_(handles->release()) {}
+AutoAttributeList::AutoAttributeList(std::unique_ptr<uint8_t[]>&& data,
+                                     HANDLE stdin_h, HANDLE stdout_h,
+                                     HANDLE stderr_h)
+  : data_(std::move(data)) {}
```

Comment:
Forgot to initialize `handles_` with `stdin_h` and friends here?

Reply:
If initializing `union` is too cumbersome, can revisit this:

> In that case, maybe just store them as HANDLE handles_[3] (so you can't have other member variables in between) and memcpy in constructor?

Reply:
> Forgot to initialize `handles_` with `stdin_h` and friends here?

Gah, I'm a dummy. Thanks.

Nah, union is fine. I know it adds boilerplate but I think it makes it more obvious what the memory layout is and the fact that it has to be consecutive.